src/main/java/graphql/nadel/engine/FieldIdUtil.java

```diff
@@ -0,0 +1,116 @@
+package graphql.nadel.engine;
+
+import graphql.language.Field;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static graphql.Assert.assertNotNull;
+import static graphql.Assert.assertTrue;
+
+public class FieldIdUtil {
+
+    private static final String NADEL_FIELD_ID = "NADEL_FIELD_ID";
+
+    private static class NadelInfo implements Serializable {
+        private final String id;
+        private final boolean rootOfTransformation;
+
+        public NadelInfo(String id, boolean rootOfTransformation) {
+            this.id = id;
+            this.rootOfTransformation = rootOfTransformation;
+        }
+    }
+
+    public static List<String> getRootOfTransformationIds(Field field) {
+        String serialized = field.getAdditionalData().get(NADEL_FIELD_ID);
+        if (serialized == null) {
+            return Collections.emptyList();
+        }
+        List<NadelInfo> nadelInfos = readNadelInfos(serialized);
+
+        return nadelInfos.stream()
+                .filter(nadelInfo -> nadelInfo.rootOfTransformation)
+                .map(nadelInfo -> nadelInfo.id)
+                .collect(Collectors.toList());
+    }
+
+    public static List<String> getFieldIds(Field field) {
+        String serialized = field.getAdditionalData().get(NADEL_FIELD_ID);
+        if (serialized == null) {
+            return Collections.emptyList();
+        }
+        List<NadelInfo> nadelInfos = readNadelInfos(serialized);
+        return nadelInfos.stream()
+                .map(nadelInfo -> nadelInfo.id)
+                .collect(Collectors.toList());
+    }
```

Comment:
This is almost the same code twice.  Can we make a helpere method

Reply:
done